### DISCUSSION TITLE
Handle multiple outputs

### DISCUSSION THREAD

**Discussion by muyuuuu** [Lightning-AI/pytorch-lightning#11914] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hello, I want to implement a semi-learning algorithm by pytorch lighting, here is torch training process:

```py
student = get_model()
teacher = get_model()

...

x_semi = get_semi_data()
x, label = get_data()

y_semi = teacher(x_semi).detach()
y1 = student(x)
y2 = student(x_semi)

cost = loss(y1, label) + loss(y_semi, y2)
cost.backward()

for i, j in zip(stduent, teacher):
    j.data.mul_(0.9).add_(0.1, i.data)
```

teacher model generates labels for students model. When I read docs of pytorch lighting, I can't find some good idea how to implement it in `foward` method or others. Can you help me? thank you.
